src/system/SelectableSystem.lua

```diff
@@ -70,6 +82,55 @@ function SelectableSystem:centerCursor ( gameObject )
 	end
 end
 
+function SelectableSystem:displayPathOverlay (map)
+	if not self.path then return end
+	self:clearPathOverlay()
+	for i, pathpoint in ipairs(self.path) do
+		local s = self.registry:getIdsByPool("Addressable", function(obj)
```

Comment:
@thimovss Could go from O(n^2) to O(n) operations here: getIdsByPool returns an array, and could make this much more performant.